unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,217 @@
+        redis_client_auto_register._blocking_command_responses = \
+            {"register": {"transaction_id": f"{TRANSACTION_ID}"}}
+        redis_client_auto_register._on_register(message)
+
+    def test_on_unregister(self, redis_client_auto_register):
+        """assigning the correct arguments to blocking command response and
+        checking by calling on_unregister function with correct message"""
+        message = {"data": {"device_uuid": TRANSACTION_ID, "transaction_id": TRANSACTION_ID,
+                            "response": "success"}}
+        message["data"] = json.dumps(message["data"])
+        redis_client_auto_register._blocking_command_responses \
+            = {"unregister": {"transaction_id": f"{TRANSACTION_ID}"}}
+        redis_client_auto_register._on_unregister(message)
+
+    def test_on_unregister_throws_exception(self, redis_client_auto_register):
+        """checks if response is not success should throw exception """
+        with pytest.raises(RedisAPIException,
+                           match=f"Failed to unregister from market {AREA_ID}."
+                                 "Deactivating connection."):
+            message = {"data": {"device_uuid": TRANSACTION_ID, "transaction_id": TRANSACTION_ID,
+                                "response": "unsuccessful"}}
+            message["data"] = json.dumps(message["data"])
+            redis_client_auto_register._blocking_command_responses \
+                = {"unregister": {"transaction_id": f"{TRANSACTION_ID}"}}
+            redis_client_auto_register._on_unregister(message)
+
+    def test_on_event_or_response(self, redis_client_auto_register):
+        """assigning the correct arguments to blocking command response and
+        checking by calling on_event_or_response function with correct message"""
+        message = {"data": {"device_uuid": TRANSACTION_ID,
+                            "transaction_id": TRANSACTION_ID}}
+        message["data"] = json.dumps(message["data"])
+        redis_client_auto_register._on_event_or_response(message)
+
+    def test_select_aggregator(self, redis_client_auto_register):
+        """checks if redis_db.publish is called with correct arguments"""
+        aggregator_uuid = str(uuid.uuid4())
+        self.redis_client = redis_client_auto_register
+        self.redis_client.redis_db.publish = MagicMock()
+        self.redis_client.area_uuid = AREA_ID
+        self.redis_client.select_aggregator(aggregator_uuid, is_blocking=False)
+        transaction_id = self.redis_client._transaction_id_buffer[0]
+        data = {"aggregator_uuid": aggregator_uuid,
+                "device_uuid": AREA_ID,
+                "type": "SELECT",
+                "transaction_id": transaction_id}
+        self.redis_client.redis_db.publish.assert_called_once_with("aggregator", json.dumps(data))
+
+    def test_select_aggregator_self_area_uuid_none_throws_exception(self, redis_client_auto_register):
```

Comment:
the function name and the docstring do not fit to what is done in there.

Reply:
Here you are testing the case when you havent registered the device to the aggregator. This is shown by `self.area_uuid=None` but please add this at least to the docstring. Thanks!